tap_woo/streams.py

```diff
@@ -591,29 +330,45 @@ class SubscriptionsStream(wooStream):
         th.Property("end_date_gmt", th.DateTimeType),
         th.Property("resubscribed_from", th.StringType),
         th.Property("resubscribed_subscription", th.StringType),
-        th.Property("removed_line_items", th.ArrayType(th.IntegerType))
+        th.Property("removed_line_items", th.ArrayType(th.IntegerType)),
     ).to_dict()
 
+    def post_process(self, row: dict, context: dict | None = None) -> dict | None:
+        list_date_fields = [
+            'trial_end_date_gmt',
+            'next_payment_date_gmt',
+            'last_payment_date_gmt',
+            'cancelled_date_gmt',
+            'end_date_gmt'
+        ]
+        for date_time_field in list_date_fields:
+            if date_time_field in row and not row[date_time_field]:
+                del row[date_time_field]
+        return super().post_process(row, context)
+
     def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
         """Return a context dictionary for child streams."""
         return {
-                "subscription_id": record["id"],
-            }
+            "subscription_id": record["id"],
+        }
 
 
 class SubscriptionOrdersStream(wooStream):
     name = "subscription_orders"
     path = "/subscriptions/{subscription_id}/orders"
-    primary_keys = ["subscription_id"]
+    primary_keys = ["order_id"]
```

Comment:
Is this what was causing the `state` mess?  Regardless, the `order_id` probably makes more sense as the primary key, since this should be a many to one relationship, where the order id will be unique (as the many) rather than the subscription id (as the one).

Reply:
Nevermind, i see the addition of `state_partitioning_keys` - I assume that's what did it.  I'm still curious how that works/should work 🤔

Reply:
yes, it was the `state_partitioning_keys` that stores the values in the state. I think it's used in case you want to check if some state has already been included. But it can create a mess in the state dict if we have too much data.